### PR TITLE
Fixes for build, recursive resolver fix

### PR DESCRIPTION
**Pull request by sslb**
Hey,

Nice project, good job! I have resolved a couple of issues, implemented maven build and resolved an issue with one of the recursive functions resolving nested properties.

Also changed some code to be more 'java'. If you are happy to pull my changes then I'll destroy my fork and happy to contribute to your project. If not, that's OK, I'll maintain the fork separately.

Szabolcs

### CODE REVIEW

**Comment by esh-b** [comment]
Hey @sslb, thanks for the contribution. I will review the changes and merge your request.

File: src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java [hunk 1]
```diff
@@ -115,44 +129,43 @@ public String getBasePath() {
...
 
         //Incase the schemes is mentioned in the RAML, use it. Else parse it from the API URL
-        if(raml.getProtocols().size() > 0) {
-            List<String> protoList = new ArrayList<>();
-            for(Protocol p : raml.getProtocols()) {
-                protoList.add(p.toString().toLowerCase());
-            }
-            swaggerJSON.put(SCHEMES_PARAM_KEY, protoList);
+        if (raml.getProtocols().size() > 0) {
+            swaggerJSON.put(SCHEMES_PARAM_KEY, raml.getProtocols().stream().map(p -> p.toString().toLowerCase()).collect(Collectors.toList()));
```
Comment (esh-b) [inline src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java]: I would suggest to add a line of comment for the users to understand easily in places where code seems pretty complex.

**Comment by esh-b** [comment]
Hey @sslb, thanks for making the code "**javaish**" :). But, few queries.. 
1) Can I know what are the test cases (raml files) that you used to validate your new code? Also, make sure you have updated the **readme** accordingly.
2) Also, I would suggest to add comments for lines where maps or some complex functions are used so that it's easier for the reader.

File: src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java (hunk 1 above)
Comment (sslb) [reply src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java]: Sure. This is the Java stream API introduced in Java8 to simplify work with collections. Will comment.

Comment (esh-b) [reply src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java]: Oh..then, why dont we go with the earlier one so as to ensure compatibility for Java 7?

Comment (sslb) [reply src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java]: Sorry for the late reply. I thought this is a standalone tool, not a library therefore compatibility is not a question.

Java 7 is out of life and no longer supported. Java 9 is already out. Whereas I embrace backwards compatibility, sometimes there's no reason for it. Java 8+ comes with lots of performance benefit and a functional interface.The map() function is part of this and reduces unnecessary verbosity.

At the end of the day this is your project and you decide how you want to do it, all I'm saying that Java7 should not be a goal for this project in my opinion.

Comment (esh-b) [reply src/main/java/org/macrulez/utils/raml2swagger/RAMLtoSwagger.java]: Actually, even I created it as a standalone tool. But, I am not sure whether users are interested in it as a standalone tool or as a lib. So, for now, I will merge your request. And if the users' needs change, we will either rollback or add that as a submodule. 
So, final check..are all the test cases' outputs equal with the outputs of the previous code? I just want to make sure the project is working EOD.

**Pull request merged**